optimum/habana/transformers/trainer_seq2seq.py

```diff
@@ -151,7 +151,11 @@ def evaluate(
 
         gen_kwargs = gen_kwargs.copy()
         if gen_kwargs.get("max_length") is None and gen_kwargs.get("max_new_tokens") is None:
-            gen_kwargs["max_length"] = self.args.generation_max_length
+            if self.args.generation_max_new_tokens:
+                gen_kwargs["max_new_tokens"] = self.args.generation_max_new_tokens
+                self.model.generation_config.max_new_tokens = self.args.generation_max_new_tokens
+            else:
+                gen_kwargs["max_length"] = self.args.generation_max_length
```

Comment:
I don't think this is necessary anymore after #293.

Reply:
Again, couldn't find anything that propagates max_new_tokens #293